packages/mui-base/src/ModalUnstyled/ModalUnstyled.tsx

```diff
@@ -40,17 +46,27 @@ const defaultManager = new ModalManager();
 /**
  * Modal is a lower-level construct that is leveraged by the following components:
  *
- * - [Dialog](/material-ui/api/dialog/)
- * - [Drawer](/material-ui/api/drawer/)
- * - [Menu](/material-ui/api/menu/)
- * - [Popover](/material-ui/api/popover/)
+ * *   [Dialog](/material-ui/api/dialog/)
+ * *   [Drawer](/material-ui/api/drawer/)
+ * *   [Menu](/material-ui/api/menu/)
+ * *   [Popover](/material-ui/api/popover/)
  *
  * If you are creating a modal dialog, you probably want to use the [Dialog](/material-ui/api/dialog/) component
  * rather than directly using Modal.
  *
  * This component shares many concepts with [react-overlays](https://react-bootstrap.github.io/react-overlays/#modals).
+ *
+ * Demos:
+ *
+ * - [Unstyled Modal](/base/react-modal/)
+ *
+ * API:
+ *
+ * - [ModalUnstyled API](/base/api/modal-unstyled/)
  */
-const ModalUnstyled = React.forwardRef(function ModalUnstyled(props, ref) {
+const ModalUnstyled = React.forwardRef(function ModalUnstyled<
+  BaseComponentType extends React.ElementType = ModalUnstyledTypeMap['defaultComponent'],
+>(props: ModalUnstyledProps<BaseComponentType>, forwardedRef: React.Ref<any>) {
```

Comment:
It's not necessary to define ModalUnstyled as a generic function. The generic signature will be lost anyway because it's wrapped in a `forwardRef`.

Instead cast the whole `forwardRef(..)` result to an `OverridableComponent<ModalUnstyledTypeMap>`. See how it's done in a ButtonUnstyled for example: https://github.com/mui/material-ui/blob/31eeb83281b0edd3bda55c6dbf926d5de2c50d62/packages/mui-base/src/ButtonUnstyled/ButtonUnstyled.tsx#L103